Framework/Core/include/Framework/AnalysisTask.h

```diff
@@ -150,14 +150,8 @@ struct OutputObj {
   {
     static_assert(std::is_base_of_v<TNamed, T>, "You need a TNamed derived class to use OutputObj");
     header::DataDescription desc{};
-    if (object == nullptr) {
-      strncpy(desc.str, "__OUTPUTOBJECT__", 16);
-    } else {
-      // FIXME: for the moment we use GetTitle(), in the future
-      //        we should probably use a unique hash to allow
-      //        names longer than 16 bytes.
-      strncpy(desc.str, object->GetTitle(), 16);
-    }
+    memset(&desc.str, '_', 16);
+    std::memcpy(&desc.str, label.c_str(), 16);
```

Comment:
+1 for the `memset`, are you sure it's not simply `desc.str`, without `&` though? 

I would still use `object->GetName()` rather than `label`, because the binding could be the same for a variety of algorithms (e.g. "histo") while the `object->GetName()` should better reflect the contents ("e.g. phiDistribution").

Reply:
I am not sure about `&`, so I'll have to test it. Regarding the label, `OutputObj` can be only constructed with a label set, either explicitly or from the pre-made object. If the object is added later, that object's name will be reset to label since down the line we will receive only the object itself and need to match it.

Reply:
Indeed `&` is not needed.